docs/dev/sql-nested-function-select-clause.md

```diff
@@ -0,0 +1,484 @@
+{
+  "mappings": {
+    "properties": {
+      "message": {
+        "type": "nested",
+        "properties": {
+          "info": {
+            "type": "keyword",
+            "index": "true"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+**Dataset:**
+```json
+{"index":{"_id":"1"}}
+{"message":{"info":"a"}}
+```
+
+**Query:**
+`SELECT nested(message.info) FROM nested_objects;`
+
+**Response:**
+```json
+{
+    "schema": [
+        {
+            "name": "nested(message.info)",
+            "type": "keyword"
+        }
+    ],
+    "datarows": [
+        [
+            "a"
+        ]
+    ],
+    "total": 1,
+    "size": 1,
+    "status": 200
+}
+```
+
+A basic nested function in the SELECT clause and output DSL pushed to OpenSearch. This example queries the `nested` object `message` and the inner field `info` to return all matching inner fields values.
+- `SELECT nested(message.info, message) FROM nested_objects;`
+```json
+{
+    "from": 0,
+    "size": 200,
+    "timeout": "1m",
+    "query": {
+        "bool": {
+            "filter": [
+                {
+                    "bool": {
+                        "must": [
+                            {
+                                "nested": {
+                                    "query": {
+                                        "match_all": {
+                                            "boost": 1.0
+                                        }
+                                    },
+                                    "path": "message",
+                                    "ignore_unmapped": false,
+                                    "score_mode": "none",
+                                    "boost": 1.0,
+                                    "inner_hits": {
+                                        "ignore_unmapped": false,
+                                        "from": 0,
+                                        "size": 3,
+                                        "version": false,
+                                        "seq_no_primary_term": false,
+                                        "explain": false,
+                                        "track_scores": false,
+                                        "_source": {
+                                            "includes": [
+                                                "message.info"
+                                            ],
+                                            "excludes": []
+                                        }
+                                    }
+                                }
+                            }
+                        ],
+                        "adjust_pure_negative": true,
+                        "boost": 1.0
+                    }
+                }
+            ],
+            "adjust_pure_negative": true,
+            "boost": 1.0
+        }
+    },
+    "_source": {
+        "includes": [
+            "message.info"
+        ],
+        "excludes": []
+    }
+}
```

Comment:
There's a lot going on in this DSL query and it's hard to understand which parts were added due to nested expression.

Is it possible to delete any of the properties? For example,  by default `adjust_pure_negative` is set to `true` by OpenSearch. I expect the other boolean properties are being set to their default values as well. Even if explain endpoint shows these values, as a reader it will help if they are not there.